### ISSUE TITLE
Can't change between 'maps' with different zoom levels (in browser).

### ISSUE THREAD

**Issue by Reg-Ed** [webbukkit/dynmap#2179]
I've just started using DynMap (as a refugee from The Overviewer which has been pronounced dead by one of their staff) and I'm very impressed so far.  There have been a few glitches in the rendering process (ours is a very large map) but nothing I haven't been able to overcome.

There is a problem, however, when changing between map views in the browser, when the maps have different amounts of zoom levels and you try to change between a map at one zoom level, and another that doesn't have that zoom level.

As an example, I have two maps for my main overworld - flat and surface, both set to extrazoomout 3.  Flat is rendered at lowres with mapzoomin set to 1, total 7 zoom levels.  Surface is rendered at hires with mapzoomin set to 0, total 8 zoom levels.  Everything is fine (and I can switch between maps no problem) until I'm on 'surface' at maximum zoom (8) and try switching to flat.  'Flat' has no level 8 zoom so the map freezes on 'surface' (blurry as though it were trying to change back to level 7) and won't load any more tiles from the server.  I have to refresh the browser to fix it.

Switching to my End maps (flat and surface) also causes similar problems as they both have a different set of parameters leading to different amounts of zoom levels.  In some situations I can switch from overworld to end but the overworld markers persist and show on the end map.  Weird.

Anyway, tl;dr how about making it so that if you try to switch from one map/zoom level to a map that doesn't have that zoom level, it switches to the closest zoom level that map does have?

**Comment by Pisi-Deff** [comment]
> how about making it so that if you try to switch from one map/zoom level to a map that doesn't have that zoom level, it switches to the closest zoom level that map does have?

So I did some research and, funnily enough, that exact logic exists and happens to be what is triggering the bug as a side effect.

I'm relatively certain that it can be fixed by replacing the contents of the file `/plugins/dynmap/web/js/markers.js` with this:
https://gist.github.com/Pisi-Deff/4f44052954f5a5132f704ae2964a5760

The only change is adding this into the conditional on line 134: ` && marker.our_layer`

@Reg-Ed 
Please let me know if it solves it for you as well. If it does, I will create a pull request for the change to be included in dynmap's source.

**Comment by Reg-Ed** [comment]
@Pisi-Deff Thanks for responding.  Having made your suggested change, my original problem has been fixed, I can now switch between maps with different zoom levels and see the expected behaviour.

However, none of my markers now load and I get an error after the page has loaded -  "Error loading js/markers.js".

**Comment by Pisi-Deff** [comment]
Hmm... odd.

Definitely no syntax issues in the file... and it works fine on my own server.
Any chance you could post a link to your dynmap?

**Comment by Reg-Ed** [comment]
Is there any way to send it to you privately?

**Comment by Pisi-Deff** [comment]
You can email me at eerik46@gmail.com
